pkg/fs/fs.go

```diff
@@ -0,0 +1,47 @@
+package fs
+
+import (
+	"io"
+
+	"machinerun.io/atomfs/pkg/common"
+	"machinerun.io/atomfs/pkg/erofs"
+	"machinerun.io/atomfs/pkg/squashfs"
+	"machinerun.io/atomfs/pkg/verity"
+)
+
+type Filesystem interface {
+	// stacker
+	Make(tempdir string, rootfs string, eps *common.ExcludePaths, verity verity.VerityMetadata) (io.ReadCloser, string, string, error)
+	ExtractSingle(squashFile string, extractDir string) error
+	// atomfs
+	Mount(squashfs, mountpoint, rootHash string) error
```

Comment:
we probably need some common name besides squashfs, like maybe atomfs since we can compress the data in different formats.